### ISSUE TITLE
SA1600 is not raised for record primary constructors that are missing param tags

### ISSUE THREAD

**Issue by SapiensAnatis** [DotNetAnalyzers/StyleCopAnalyzers#3780]
Today, the following code will not cause any Stylecop diagnostics to be raised:

```csharp
/// <summary>
/// Record.
/// </summary>
public record MyRecord(int Parameter);
```

This is in spite of the missing `<param>` tag for `Parameter`. Stylecop should report a diagnostic unless this is included:

```csharp
/// <summary>
/// Record.
/// </summary>
/// <param name="Parameter">Parameter.</param>
public record MyRecord(int Parameter);
```

Because `Parameter` declares a property as well as a constructor parameter, it was agreed in #3770 that it should report SA1600 instead of the usual SA1611 for missing `<param>` tags.

**Comment by sharwell** [comment]
I think there are some versions of Roslyn where callbacks execute twice, resulting in duplicate diagnostics. You can include the diagnostic twice in the expected results and I can investigate further.
